chain/index/chain-indexing-overview-for-rpc-providers.MD

```diff
@@ -0,0 +1,202 @@
+# ChainIndexer Documentation for RPC Providers
+  
+## Introduction
+
+This document is for RPC providers and node operators who serve RPC requests walking through the configuration changes, migration flow and operations/maintenance work needed to enable, backfill and maintain the `ChainIndexer`.
+
+**Note: If you are a Storage Provider or node operator who does not serve RPC requests, you can skip this document as the `ChainIndexer` is already disabled by default**. 
+ 
+
+## ChainIndexer Config
+### Enablement
+
+The following must be enabled on an RPC node before starting as they are disabled by default:
+
+```toml
+[ChainIndexer]
+# Enable the ChainIndexer. 
+  EnableIndexer = true 
+
+[Fevm]
+# Enable the ETH RPC APIs.
+  EnableEthRPC = true
```

Comment:
what happens if i have a node that has enable eth rpc set to true, and i restart the node with the new release that with this PR without changing the default config for EnableIndexer (which will be set to false)?

Reply:
@jennijuju You will still store event logs in your chain state(so they can be indexed later when you enable the Indexer without re-computing the tipset) but none of the txns,events and messages will be indexed which means that RPC calls will either fail or be really slow.

You can then later turn on `ChainIndexer` and do a back-fill to get that state into the Index.

Reply:
hrm so the migration is mandatory..? i assume you cant keep both system for a while and deprecate the old indexing system w/o duplicating a lot of code? Then what's the point of introducing ` EnableIndexer`? cuz it seems like enable eth rpc  + enable actor event -> must be enableiindexer?

Reply:
I think that's a good/valid question @jennijuju .

@aarshkshah1992 : is there a usecase to have `EnableEthRC == true && EnableActorEventsAPI == true` but `EnableIndexer == false`?  if not, then it seems like `ChainIndexer.EnableIndexer` could be removed and implicitly be enabled.

(That said, I assume this is useful to have a config flag, either for clarity and/or testing purposes.)

Reply:
Last I looked I think ChainIndexer was doing roughly the same as we have for now wrt this: you can `EnableEthRPC` which will enable `RealTimeFilterAPI` and enable "HistoricFilterAPI", which I think is the same for `EnableActorEventsAPI`. for you, but you can run with `DisableHistoricFilterAPI` set to `true`. This means that your node will extract events from tipset execution and serve them up with "real-time" APIs ("subscribe" and the eth "filter" APIs) but you can't ask for anything historic.

I'm not entirely convinced the cost of the confusion about this is worth allowing for that use-case. It seems like users don't like the eth subscribe and filter APIs and prefer to be explicit with getting logs for certain epochs and polling. I don't think I've even seen anyone use subscribe or filter APIs in practice! Not that my experience is broad.

Reply:
Either way, this doc should clearly outline the behaviour, because it's confusing and is worth some prose.

Reply:
( @rvagg in the last couple month you have been working with indexer providers and explorer like tool builders the most and they need all events. That said, dapp builders , often only care about the event their contracts emits and they use subscribe mostly; another filecoin example is client contract deals - SP softwares will filter for specific deal proposal events and react to those accordingly. So yah there is definitely an important user base for those api

Reply:
I'm moving this to https://github.com/filecoin-project/lotus/pull/12600